src/Services/InventoryUpdateService.php

```diff
@@ -184,13 +185,34 @@ public function sync(bool $fullInventory = false): array
       // TODO: consider failing out of one item / one page instead of failing the whole sync
       $externalLogs->addInventoryLog('Inventory: ' . $e->getMessage(), 'inventoryFailed' . ($fullInventory ? 'Full' : ''), 1, 0, false);
 
+      $exceptionType = get_class($e);
+      $msg = $e->getMessage();
+      $stack = $e->getTrace();
+      $lenStack = count($stack);
+      $lenMsg = strlen($msg);
+
+      if ($lenStack > 2)
+      {
+        // truncate the stack to avoid PM saying the log message is too large
+        $stack = array_slice($stack, 0, 2);
+        $stack[] = '...';
+      }
+
+      if ($lenMsg > 64)
+      {
+        // message is over 300k here!
+        $msg = substr($msg, 0, 64);
```

Comment:
not an issue: I'm curious if there'd be a way to encapsulate message length within the logger contract classes

Reply:
yeah that's some temporary stuff I need to clean out when I fix the merge conflicts.
Ideally, our wrapper for the PM logger can detect things that are too large and truncate and/or chunk the message.

Reply:
created https://projecthub.service.csnzoo.com/browse/EM-100/ to capture that effort